src/main/java/org/team1540/base/util/AsyncCommand.java

```diff
@@ -0,0 +1,131 @@
+package org.team1540.base.util;
+
+import edu.wpi.first.wpilibj.Notifier;
+import edu.wpi.first.wpilibj.command.Command;
+import org.jetbrains.annotations.Contract;
+
+/**
+ * A {@link Command} that spawns off a new {@link Notifier} on initialization and stops it. That
+ * thread will call the {@link #runPeriodic()} ()} method at a user-specified interval with precise
+ * timings.
+ *
+ * @see Thread
+ */
+public abstract class AsyncCommand extends Command {
+
+  private long interval;
+  private boolean finished = false;
+
+  private Notifier notifier;
+
+  /**
+   * Constructs a new {@code AsyncCommand} with a preset periodic interval.
+   *
+   * @param interval The interval between {@link #run()} calls, in milliseconds.
+   */
+  public AsyncCommand(long interval) {
+    this.interval = interval;
+  }
+
+  /**
+   * Runs the action of this command.
+   * <p>
+   * This is equivalent to the {@link #execute()} method in normal commands. Note that this method
+   * will be called in a seperate thread, and as such any public methods it calls, fields it uses,
+   * etc. MUST be thread-safe.
+   */
+  protected void runPeriodic() {
```

Comment:
Why is this not abstract?

Reply:
Much like the standard WPILib command methods, you won't need _all_ the methods in every command that you run; therefore, this is the default implementation.

Reply:
Might be worth noting that the default implementation is empty in the JavaDoc.